docs/contributing/guidelines-for-writing-upgrade.md

```diff
@@ -0,0 +1,20 @@
+# Guidelines for writing UPGRADE.md
+
+Keep in mind that upgrade instructions are written for users that do not understand our system well, so more clear they are, more useful they are.
+
+* Our users work in a clone of project-base and even they upgrade, their project-base is not upgraded.
```

Comment:
I am not sure what you meant to write - probably "...even though they do/perform the upgrade, their project-base..." or "...even when they do/perform the upgrade, their..."